src/Compilers/CSharp/Portable/Binder/SwitchExpressionArmBinder.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal class SwitchExpressionArmBinder : Binder
+    {
+        private readonly SwitchExpressionArmSyntax _arm;
+        private readonly ExpressionVariableBinder _armScopeBinder;
+        private readonly SwitchExpressionBinder _switchExpressionBinder;
+
+        public SwitchExpressionArmBinder(SwitchExpressionArmSyntax arm, ExpressionVariableBinder armScopeBinder, SwitchExpressionBinder switchExpressionBinder) : base(armScopeBinder)
+        {
+            this._arm = arm;
+            this._armScopeBinder = armScopeBinder;
+            this._switchExpressionBinder = switchExpressionBinder;
+        }
+
+        internal override BoundSwitchExpressionCase BindSwitchExpressionArm(SwitchExpressionArmSyntax node, DiagnosticBag diagnostics)
+        {
+            Debug.Assert(node == _arm);
+            var caseBinder = this.GetBinder(node);
+            var hasErrors = _switchExpressionBinder.SwitchGoverningExpression.HasErrors;
```

Comment:
Is it necessary for `hasErrors` to be set for this case if the governing expression `HasErrors` is set? Couldn't the two be independent?

Reply:
The patterns of the switch expression's arms are bound using the type of the governing expression, so an error in the switch expression is likely to lead to spurious errors in the patterns. However, I think it would be sufficient to set the hasError flag when the switch governing expression's *type* is an error type, as other kinds of errors do not result in cascaded diagnostics.

---
In reply to: [169863147](https://github.com/dotnet/roslyn/pull/23357#discussion_r169863147) [](ancestors = 169863147)